test/jdk/javax/swing/JFileChooser/FileSystemView/SystemIconTest.java

```diff
@@ -72,9 +72,20 @@ static void negativeTests() {
 
     static void testSystemIcon(File file, boolean implComplete) {
         int[] sizes = new int[] {16, 32, 48, 64, 128};
+        if (!file.exists() || !file.canRead()) {
+            System.err.println("File " + file.getAbsolutePath() + " is inaccessible");
+            return;
+        }
+
         for (int size : sizes) {
-            ImageIcon icon = (ImageIcon) fsv.getSystemIcon(file, size, size);
```

Comment:
Probably we can wrap the  "UIManager.getIcon" in the "ImageIcon"?

Reply:
> Probably we can wrap the "UIManager.getIcon" in the "ImageIcon"?

Why? In some LaFs that might be a procedural image that generates it content every time paint() is called on it - i do not think wrapping it in the ImageIcon will solve anything.

Reply:
If it is generated every time, then it will be wrapped every time as well. The difference is that it will not be needed to check the type of the icon. Even now, on Windows it is necessary to check is the returned object ImageIcon or not.

Reply:
> Even now, on Windows it is necessary to check is the returned object ImageIcon or not.

Yes but since API states that returned type is an Icon it should be done anyways - because we might wrap up Icon in ImageIcon locally but we can't force any 3rd party Java implementation to do the same so there will be requirement to check the returned type - unless we change the API to always return ImageIcon or MultiResolutionImageIcon or something even more complicated.